org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/refactoring/code/flow/FlowAnalyzer.java

```diff
@@ -1119,8 +1120,10 @@ private void endVisitMethodInvocation(ASTNode node, ASTNode receiver, List<Expre
 		}
 		MessageSendFlowInfo info = createMessageSendFlowInfo();
 		setFlowInfo(node, info);
-		for (Iterator<Expression> iter = arguments.iterator(); iter.hasNext();) {
-			Expression arg = iter.next();
+		for (Expression arg : arguments) {

```

Comment:
Any additional changes in jdt.ls for this class? If no, we should reuse `FlowAnalyzer.java` directly from jdt.core.manipulation?

Reply:
Looks like you moved FlowAnalyzer into jdt.core.manipulation https://git.eclipse.org/r/c/jdt/eclipse.jdt.ui/+/149882/5/org.eclipse.jdt.core.manipulation/core+extension/org/eclipse/jdt/internal/corext/refactoring/code/flow/FlowAnalyzer.java for 4.14.

A comparison shows they're nearly identical : https://pastebin.com/vpNPWcD4

Reply:
I have removed the org.eclipse.jdt.ls.core.internal.corext.refactoring.code.flow package.

Reply:
Nice.In fact, the main differences found are in FlowAnalyzer, and InputFlowAnalyzer, and when looking through JDT, those differences are introduced specifically to handle https://bugs.eclipse.org/bugs/show_bug.cgi?id=488942 & https://bugs.eclipse.org/bugs/show_bug.cgi?id=545161 .